### PR TITLE
rollback ddl stmts on failure

### PR DESCRIPTION
**Pull request by teros0**

### CODE REVIEW

File: version.go [hunk 1]
```diff
@@ -254,30 +254,33 @@ func (db *DB) UpdateSchema(execDefault bool) error {
...
 				_, err := con.Exec(query)
 				if err != nil {
-					return err
+					return fmt.Errorf("exec query: (%s)\nerr: (%s)", query, err)
 				}
 			}
 			_, err := Wrap(con).Insert(&SchemaMigration{Version: version, Created: time.Now().UTC()})
 			if err != nil {
-				return err
+				return fmt.Errorf("insert to mw_sqchema_migration: %s", err)
```
Comment (wrunk) [inline version.go]: Typo

File: version.go [hunk 2]
```diff
@@ -342,6 +345,12 @@ func (db *DB) ExecuteMigration(version string) error {
 	}
 
 	if err := db.execInTx(version, migrationFunc); err != nil {
+		errLog := NewLog(actionUpdate, fmt.Sprintf("while applying migration: %s", err), version)
```
Comment (wrunk) [inline version.go]: Will this sound like an error message on failure? I know we have the success=false, but wondering if this will sound like an error

Comment (teros0) [reply version.go]: Think so, example:
`while applying migration: exec query: (ALTER TABLE `user_profile2020` ADD COLUMN `zipcode` varchar(10) NULL AFTER `mobile_number`)
err: (Error 1060: Duplicate column name 'zipcode')`

File: version.go (hunk 1 above)
Comment (teros0) [reply version.go]: fixed

**Pull request merged**